### PR TITLE
Add smrt intify op

### PR DESCRIPTION
**Pull request by MasterDuke17**
This will be needed for an about-to-land NQP PR that makes NQP default to using int instead of num behavior for variables and ops.

### CODE REVIEW

File: src/core/interp.c [hunk 1]
```diff
@@ -6376,7 +6385,7 @@ void MVM_interp_run(MVMThreadContext *tc, void (*initial_invoke)(MVMThreadContex
             OP(DEPRECATED_31):
                 MVM_exception_throw_adhoc(tc, "The syncpipe op was removed in MoarVM 2017.07.");
             OP(DEPRECATED_32):
-                MVM_exception_throw_adhoc(tc, "The close_fhi op was removed in MoarVM 2017.07.");
+                MVM_exception_throw_adhoc(tc, "The foobar op was removed in MoarVM 2017.07.");
```
Comment (jnthn) [inline src/core/interp.c]: This change looks accidental.

Comment (MasterDuke17) [reply src/core/interp.c]: Woops, force pushed a fix.

File: src/core/coerce.c [hunk 2]
```diff
@@ -378,6 +378,42 @@ void MVM_coerce_smart_numify(MVMThreadContext *tc, MVMObject *obj, MVMRegister *
         else
             MVM_exception_throw_adhoc(tc, "cannot numify this");
     }
+    {
+        MVMint64 t = (MVMint64)res_reg->n64;
+        if ((MVMnum64)t != res_reg->n64) {
+            //fprintf(stderr, "got an actual non-int value %g\n", res_reg->n64);
+        }
+    }
```
Comment (jnthn) [inline src/core/coerce.c]: This block looks like a dead code debug leftover; I must have missed it in the first review, sorry.

Comment (MasterDuke17) [reply src/core/coerce.c]: Argh, can't believe I committed that. Anyway, force pushed another update.

**Comment by MasterDuke17** [comment]
Rebased to HEAD on master.

File: src/spesh/optimize.c [hunk 3]
```diff
@@ -929,19 +929,21 @@ static void optimize_repr_op(MVMThreadContext *tc, MVMSpeshGraph *g, MVMSpeshBB
 /* smrt_strify and smrt_numify can turn into unboxes, but at least
```
Comment (niner) [inline src/spesh/optimize.c]: You could mention smrt_intify here

Comment (MasterDuke17) [reply src/spesh/optimize.c]: Yeah, updated a couple other comments also and force pushed.

**Pull request merged**